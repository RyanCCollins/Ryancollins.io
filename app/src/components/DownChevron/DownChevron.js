import React, { PropTypes } from 'react';
import ChevronDown from 'react-icons/lib/fa/chevron-down';
import { Link as ScrollTo } from 'react-scroll';

const DownChevron = ({
  scrollToSection
}) => (
  <div className="intro-header__chevron">
    <ScrollTo
      to={scrollToSection}
      spy
      smooth
      offset={50}
      duration={800}
    >
      <ChevronDown className="intro-header__chevron-down" />
    </ScrollTo>
  </div>
);

DownChevron.propTypes = {
  scrollToSection: PropTypes.string.isRequired
};

export default DownChevron;
