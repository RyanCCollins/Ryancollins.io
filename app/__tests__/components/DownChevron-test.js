import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import DownChevron from '../../src/components/DownChevron/DownChevron';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<DownChevron />', () => {
  it('renders with a scroll To Section', () => {
    const component = renderer.create(
      <DownChevron scrollToSection="main" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
