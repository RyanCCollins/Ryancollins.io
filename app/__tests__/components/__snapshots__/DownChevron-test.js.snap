exports[`<DownChevron /> renders with a scroll To Section 1`] = `
<div
  className="intro-header__chevron">
  <a
    className={undefined}
    duration={800}
    offset={50}
    onClick={[Function bound ]}
    smooth={true}
    spy={true}
    to="main">
    <svg
      className="intro-header__chevron-down"
      fill="currentColor"
      fit={true}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "verticalAlign": "middle"
        }
      }
      viewBox="0 0 40 40"
      width="1em">
      <g>
        <path
          d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z" />
      </g>
    </svg>
  </a>
</div>
`;
